### PR TITLE
Fix warnings noticed by clang

### PR DESCRIPTION
**Pull request by howard0su**

### CODE REVIEW

File: src/telemetry/telem_crsf.c [hunk 1]
```diff
@@ -89,7 +89,7 @@ const char * _crsf_short_name(char *str, u8 telem)
     case TELEM_CRSF_TX_POWER: strcpy(str, _tr("Power")); break;
     case TELEM_CRSF_RX_QUALITY: strcpy(str, _tr("rxQual")); break;
     case TELEM_CRSF_TX_QUALITY: strcpy(str, _tr("txQual")); break;
-    case TELEM_CRSF_BATT_VOLTAGE: sprintf(str, _tr("Volt")); break;
+    case TELEM_CRSF_BATT_VOLTAGE: strcpy(str, _tr("Volt")); break;
```
Comment (PhracturedBlue) [inline src/telemetry/telem_crsf.c]: Lint Error:
```
92:  Almost always, snprintf is better than strcpy  
```

Comment (PhracturedBlue) [reply src/telemetry/telem_crsf.c]: google is wrong here.  snprintf is the worst option for us.  strlcpy would be better, but unneeded here.  we should disable this rule in the linter

File: src/pages/320x240x16/range_page.c [hunk 2]
```diff
@@ -37,7 +37,7 @@ static const char *startstop_dlg(struct guiObject *gui, const void *data) {
 static void _draw_page(int has_pa) {
     PAGE_ShowHeader(PAGE_GetName(PAGEID_RANGE));
     if (!has_pa) {
-        snprintf(tempstring, sizeof(tempstring), _tr("No range test possible."));
+        snprintf(tempstring, sizeof(tempstring), "%s", _tr("No range test possible."));
         GUI_CreateLabelBox(&gui->info, INFO_X, INFO_Y, 0, 0, &DEFAULT_FONT,
```
Comment (PhracturedBlue) [inline src/pages/320x240x16/range_page.c]: I think strlcpy would be better

File: src/pages/common/_debuglog_page.c [hunk 3]
```diff
@@ -28,13 +28,13 @@ static u32 changed = 0;
...
-    unsigned idx = (long)data;
+    unsigned idx = (unsigned)data;
     if (idx == 255) {
         return "";
     }
     const char *dptr = (char *)&logstr[line_pos[idx]];
     const char *w = (char *)wptr;
-    sprintf(tempstring, "%d:", (long)data);
+    sprintf(tempstring, "%d:", (unsigned)data);
     char *ptr = tempstring + strlen(tempstring);
```
Comment (PhracturedBlue) [inline src/pages/common/_debuglog_page.c]: this should probably be '%u' if using an unsigned

File: src/pages/common/_debuglog_page.c [hunk 4]
```diff
@@ -28,13 +28,13 @@ static u32 changed = 0;
 static const char *str_cb(guiObject_t *obj, const void *data)
 {
     (void)obj;
-    unsigned idx = (long)data;
+    unsigned idx = (unsigned)data;
     if (idx == 255) {
```
Comment (PhracturedBlue) [inline src/pages/common/_debuglog_page.c]: This should be uintptr_t.  you shouldn't cast a pointer to an int.

File: src/pages/common/_model_loadsave.c [hunk 5]
```diff
@@ -30,7 +30,7 @@ static int ini_handle_icon(void* user, const char* section, const char* name, co
 
 static int ini_handle_name(void* user, const char* section, const char* name, const char* value)
 {
-    long idx = (long)user;
+    int idx = (int)user;
     if(section[0] == '\0' && (strcasecmp(name, MODEL_NAME) == 0 || strcasecmp(name, MODEL_TEMPLATE) == 0)) {
```
Comment (PhracturedBlue) [inline src/pages/common/_model_loadsave.c]: what is wrong with this?  sizeof(void*) == sizeof(long) on all platforms

File: src/pages/common/_timer_page.c [hunk 6]
```diff
@@ -189,7 +189,7 @@ const char *show_timerperm_cb(guiObject_t *obj, const void *data)
   (void)obj;
   u8 idx = (long)data;
   if (idx & 0x80)
-      snprintf(tempstring, sizeof(tempstring), _tr("Set to"));
+      snprintf(tempstring, sizeof(tempstring), "%s", _tr("Set to"));
   else
```
Comment (PhracturedBlue) [inline src/pages/common/_timer_page.c]: strlcpy again

File: src/protocol/cflie_nrf24l01.c [hunk 7]
```diff
@@ -343,7 +343,7 @@ static void send_crtp_rpyt_packet()
         float pitch;
         float yaw;
         uint16_t thrust;
-    }__attribute__((packed)) cpkt;
+    }cpkt;
```
Comment (PhracturedBlue) [inline src/protocol/cflie_nrf24l01.c]: why did you remove the packed attribute.  I don't know this protocol, but it is likely the packet is strictly defined as far as its structure goes, and the compiler can't insert any padding)

File: src/protocol/inav_nrf24l01.c [hunk 8]
```diff
@@ -85,7 +85,7 @@
 
 #ifdef EMULATOR
 #define USE_FIXED_MFGID
-#define BIND_COUNT 4.360   // 6 seconds
+#define BIND_COUNT 4      // 6 seconds
 #define FIRST_PACKET_DELAY  1
```
Comment (PhracturedBlue) [inline src/protocol/inav_nrf24l01.c]: this just looks weird.  was the original a typo?

File: src/protocol/symax_nrf24l01.c [hunk 9]
```diff
@@ -41,7 +41,7 @@
 
 #ifdef EMULATOR
 #define USE_FIXED_MFGID
-#define BIND_COUNT 4.360   // 6 seconds
+#define BIND_COUNT 4      // 6 seconds
 #define FIRST_PACKET_DELAY  1
```
Comment (PhracturedBlue) [inline src/protocol/symax_nrf24l01.c]: same question as above

File: src/pages/common/_model_loadsave.c (hunk 5 above)
Comment (howard0su) [reply src/pages/common/_model_loadsave.c]: long should use %ld to printf. I am not sure if we want to do that in the following line. and idx to me int is large enough to hold idx.

File: src/protocol/cflie_nrf24l01.c (hunk 7 above)
Comment (howard0su) [reply src/protocol/cflie_nrf24l01.c]: there is risk for having packed structure on ARM platform as not ARM instruction support unaligned access. this get warned in the below code which can result unaligned access.

remove packed is not a right solution, i agree. let me tune the code a bit to manually pack the code to avoid this code accidently get broken by some compiler udpate.

File: src/protocol/inav_nrf24l01.c (hunk 8 above)
Comment (howard0su) [reply src/protocol/inav_nrf24l01.c]: I am not quite sure. @goebish can u help understand why this is a float which is expect to be an integer.

File: src/timer.c [hunk 10]
```diff
@@ -119,7 +119,10 @@ void TIMER_Power(){
 
     elevator = 2 == mode ? abs(CHAN_ReadInput(INP_THROTTLE)) : abs(CHAN_ReadInput(INP_ELEVATOR));
     new_throttle = 2 == mode ?  abs(CHAN_ReadInput(INP_ELEVATOR)) : abs(CHAN_ReadInput(INP_THROTTLE));
-    new_throttle = abs(new_throttle - throttle);
+    if (new_throttle > throttle)
```
Comment (howard0su) [inline src/timer.c]: this is warning that both are unsigned and result is unsigned as well. so we have to explict check to avoid use abs, which doesn't work.

Comment (PhracturedBlue) [reply src/timer.c]: yes, this one looks good

File: src/pages/common/_model_loadsave.c (hunk 5 above)
Comment (PhracturedBlue) [reply src/pages/common/_model_loadsave.c]: idx can be an int, but the cast should be to an equivalent sized pointer I think. intptr_t would work for the cast.(I didn't check, I assume we have both intptr_t and uintptr_t)

File: src/telemetry/telem_crsf.c (hunk 1 above)
Comment (howard0su) [reply src/telemetry/telem_crsf.c]: i would like to leave strcpy here for now. We should cleanup telemetry code later.

Comment (PhracturedBlue) [inline src/telemetry/telem_crsf.c]: Lint Error:
```
92:  Almost always, snprintf is better than strcpy  
```

File: src/protocol/inav_nrf24l01.c (hunk 8 above)
Comment (goebish) [reply src/protocol/inav_nrf24l01.c]: I suppose that was a typo, 4 is fine for EMULATOR

Comment (howard0su) [reply src/protocol/inav_nrf24l01.c]: no one knows why. but goebish believe this is fine.

File: src/telemetry/telem_crsf.c (hunk 1 above)
Comment (PhracturedBlue) [inline src/telemetry/telem_crsf.c]: Lint Error:
```
92:  Almost always, snprintf is better than strcpy  
```
